modules/polaris/src/main/ts/ephox/polaris/words/Words.ts

```diff
@@ -38,6 +38,8 @@ export interface WordsWithIndices<T> {
   readonly indices: WordIndex[];
 }
 
+const isAnAcronym = (str: string) => new RegExp(/^(?:[A-Z]\.)+$/gm).test(str);
```

Comment:
I wonder if this could cause issues in some languages. Maybe we should limit this to single letter so it doesn't pass in. `SOMETHING.` as one word. If a user types in all caps for some odd reason we shouldn't treat that as an acronym for example. Also I think this can be simplified to: `/^([A-Z]\.)+$/` since the capture group ignore shouldn't matter since we are just using test we are not extracting things also the multiline should not be needed since we only want to match on letter and dot not \n stuff anyway and since we are only matching one thing I don't think we need to make it global either.

Reply:
> I wonder if this could cause issues in some languages.
Yes this could be a matter to manage, but I think the best solution would be to check if someone complain about that, since check all the way to write could be not possible in short period

> Maybe we should limit this to single letter so it doesn't pass in. SOMETHING. as one word.
is already limited to 1 letter `SOMETHING.` doesn't pass, to make it pass you have to change it in `S.O.M.E.T.H.I.N.G.`

> Also I think this can be simplified to: /^([A-Z]\.)+$/ since the capture group ignore shouldn't matter since we are just using test we are not extracting things also the multiline should not be needed since we only want to match on letter and dot not \n stuff anyway and since we are only matching one thing I don't think we need to make it global either.

agree removing `gm` and `?:`, but I wonder if we should extend this check to the lower case too, since I don't see any reason to have it only for upper case.

Reply:
my proposal would be to change it in:
```js
const isAnAcronym = (str: string) => new RegExp(/^([A-Z]\.)+$/).test(str);
```
or
```js
const isAnAcronym = (str: string) => new RegExp(/^([a-zA-Z]\.)+$/).test(str);
```

Reply:
Right I forgot to remove the part about it checking the SOMETHING I initially misread that regex. I guess lowercase would be fine too.

Reply:
I made a [test commit](https://github.com/tinymce/tinymce/pull/9663/commits/ba6d8ee03f47863eea4eaf8ad5903c7e8e0df297) with code that should be more optimized, but sadly is way harder to read, let me know what do you think about it